File: packages/algo-go/solutions/algo-01_test.go
package solutions

import (
	"reflect"
	"testing"
)

var (
	input = [][]byte{
		{'E', 'E', 'E', 'E', 'E'},
		{'E', 'E', 'M', 'E', 'E'},
		{'E', 'E', 'E', 'E', 'E'},
		{'E', 'E', 'E', 'E', 'E'},
	}
	expected = [][]byte{
		{'B', '1', 'E', '1', 'B'},
		{'B', '1', 'M', '1', 'B'},
		{'B', '1', '1', '1', 'B'},
		{'B', 'B', 'B', 'B', 'B'},
	}
)

func TestMineSweeper(t *testing.T) {
	if !reflect.DeepEqual(expected, Minesweeper(input, []int{3, 0})) {
		t.Error("Failed")
	}
}

func TestMineSweeperDFS(t *testing.T) {
	if !reflect.DeepEqual(expected, MinesweeperDFS(input, []int{3, 0})) {
		t.Error("Failed")
	}
}

File: packages/algo-py/solutions/algo_01_test.py
import unittest
from solutions import *


input = [["E", "E", "E", "E", "E"], ["E", "E", "M", "E", "E"],
         ["E", "E", "E", "E", "E"], ["E", "E", "E", "E", "E"]]
expected = [["B", "1", "E", "1", "B"], ["B", "1", "M", "1", "B"],
            ["B", "1", "1", "1", "B"], ["B", "B", "B", "B", "B"]]


class TestAlgo_01(unittest.TestCase):

    def test(self):
        self.assertListEqual(minesweeper(input, [3, 0]), expected)

    def test_dfs(self):
        self.assertListEqual(minesweeper_dfs(input, [3, 0]), expected)


if __name__ == '__main__':
    unittest.main()
